org.eclipse.dawnsci.hdf5.test/src/org/eclipse/dawnsci/hdf5/TestPlugins.java

```diff
@@ -0,0 +1,126 @@
+/*-
+ *******************************************************************************
+ * Copyright (c) 2017 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Peter Chang - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+
+package org.eclipse.dawnsci.hdf5;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.DatasetUtils;
+import org.eclipse.january.dataset.FloatDataset;
+import org.eclipse.january.dataset.IndexIterator;
+import org.junit.Test;
+
+import hdf.hdf5lib.H5;
+import hdf.hdf5lib.HDF5Constants;
+import hdf.hdf5lib.exceptions.HDF5Exception;
+
+/**
+ * Test various filter plugins that are built into the hdf.hdf5lib
+ * plugin
+ */
+public class TestPlugins {
+
+	private final static String DATA_PATH = "/data";
+	private final static int DIM_0 = 10;
+	private final static int DIM_1 = 256;
+	private final static int DIM_2 = 512;
+	private final static long[] DATA_SHAPE = new long[] {DIM_0, DIM_1, DIM_2};
+	private final static long[] DATA_MAX_SHAPE = new long[] {HDF5Constants.H5S_UNLIMITED, DIM_1, DIM_2};
+	private final static long[] DATA_CHUNK = new long[] {1, DIM_1, DIM_2};
+
+	@Test
+	public void testZlibFilter() throws NullPointerException, HDF5Exception {
+		testFilter("test-scratch/testFilterZlib.h5", HDF5Constants.H5Z_FILTER_DEFLATE, 1);
+	}
+
+	@Test
+	public void testLZFFilter() throws NullPointerException, HDF5Exception {
+		testFilter("test-scratch/testFilterLZF.h5", 32000);
+	}
+
+	@Test
+	public void testLZ4Filter() throws NullPointerException, HDF5Exception {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.NullPointerException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)